src/body-data-wrapper.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+/* eslint-disable no-param-reassign */
+const { Request } = require('@adobe/helix-fetch');
+
+/**
+ * Extracts the _data_ from the given request. The data can be provided either as request
+ * parameters, url-encoded form data body, or a json body.
+ *
+ * Note that for post body requests, the body is consumed from the request and is no longer
+ * available.
+ *
+ * @param {Request} request The universal request
+ * @param {BodyDataOptions} [opts] Options
+ * @returns {Promise<object>} the parsed data object.
+ */
+async function getData(request, opts) {
+  const { coerceNumber, coerceInt, coerceBoolean } = opts;
+  if (/json/.test(request.headers.get('content-type'))) {
+    // eslint-disable-next-line no-return-await
+    return await request.json();
+  }
+  let data = new URL(request.url).searchParams;
+  if (/^application\/x-www-form-urlencoded/.test(request.headers.get('content-type'))) {
+    // todo: combine urlencoded and request params
+    data = new URLSearchParams(await request.text());
```

Comment:
@trieloff I think request params and form params should be combined.

Reply:
Which one would take precedence?

Reply:
combine... if both have the same value(s), merge them. 
the array order is probably POST first, params later.

Reply:
So params can override POST body?

Reply:
not override - combine. but maybe it's too magic and POST body should win.
the action still can use the `req.url` to get the request params if needed.